### PR TITLE
flet-client-flutter: update to 0.22.1, fix update script to be fully automatic

### PR DESCRIPTION
**Pull request by lucasew**
## Description of changes
Bumps for this package already "work" but always have a manual step. Now  the update script updates the lockfile too.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 310889` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>flet-client-flutter</li>
    <li>flet-client-flutter.debug</li>
    <li>flet-client-flutter.pubcache</li>
    <li>python311Packages.flet</li>
    <li>python311Packages.flet-core</li>
    <li>python311Packages.flet-core.dist</li>
    <li>python311Packages.flet-runtime</li>
    <li>python311Packages.flet-runtime.dist</li>
    <li>python311Packages.flet.dist</li>
    <li>python312Packages.flet</li>
    <li>python312Packages.flet-core</li>
    <li>python312Packages.flet-core.dist</li>
    <li>python312Packages.flet-runtime</li>
    <li>python312Packages.flet-runtime.dist</li>
    <li>python312Packages.flet.dist</li>
  </ul>
</details>

**Comment by lucasew** [comment]
Tested locally, stuff working nice.

Only caveat, that is not packaging problem actually, is that bin/flet is both the Python library and the flutter client so keep this in mind in the nixpkgs-review shell because the flet command will very likely point to the flutter client.

File: pkgs/by-name/fl/flet-client-flutter/package.nix [hunk 1]
```diff
@@ -29,6 +33,13 @@ flutter.buildFlutterApplication rec {
     "-DMIMALLOC_LIB=${mimalloc}/lib/mimalloc.o"
   ];
 
+  pubspecSource = runCommand "pubspec.lock.json" {
+      buildInputs = [ yq ];
+      inherit (flet-client-flutter) src;
+    } ''
+    cat $src/client/pubspec.lock | yq > $out
+  '';
```
Comment (wegank) [inline pkgs/by-name/fl/flet-client-flutter/package.nix:41]:
```suggestion
  passthru.pubspecSource = runCommand "pubspec.lock.json" {
      buildInputs = [ yq ];
      inherit (flet-client-flutter) src;
    } ''
    cat $src/client/pubspec.lock | yq > $out
  '';
```

Comment (lucasew) [reply pkgs/by-name/fl/flet-client-flutter/package.nix]: Yes. Nice catch. Fixed.

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 310889` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>flet-client-flutter</li>
    <li>flet-client-flutter.debug</li>
    <li>flet-client-flutter.pubcache</li>
    <li>python311Packages.flet</li>
    <li>python311Packages.flet-core</li>
    <li>python311Packages.flet-core.dist</li>
    <li>python311Packages.flet-runtime</li>
    <li>python311Packages.flet-runtime.dist</li>
    <li>python311Packages.flet.dist</li>
    <li>python312Packages.flet</li>
    <li>python312Packages.flet-core</li>
    <li>python312Packages.flet-core.dist</li>
    <li>python312Packages.flet-runtime</li>
    <li>python312Packages.flet-runtime.dist</li>
    <li>python312Packages.flet.dist</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]
The second commit message should be `flet-client-flutter: 0.22.0 -> 0.22.1`. Otherwise, LGTM.

**Review by wegank — APPROVED** [review]

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 310889` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>flet-client-flutter</li>
    <li>flet-client-flutter.debug</li>
    <li>flet-client-flutter.pubcache</li>
    <li>python311Packages.flet</li>
    <li>python311Packages.flet-core</li>
    <li>python311Packages.flet-core.dist</li>
    <li>python311Packages.flet-runtime</li>
    <li>python311Packages.flet-runtime.dist</li>
    <li>python311Packages.flet.dist</li>
    <li>python312Packages.flet</li>
    <li>python312Packages.flet-core</li>
    <li>python312Packages.flet-core.dist</li>
    <li>python312Packages.flet-runtime</li>
    <li>python312Packages.flet-runtime.dist</li>
    <li>python312Packages.flet.dist</li>
  </ul>
</details>

**Pull request merged**